src/serac/physics/integrators/displacement_hyperelastic_integrator.hpp

```diff
@@ -65,24 +67,97 @@ class IncrementalHyperelasticIntegrator : public mfem::NonlinearFormIntegrator {
                                    const mfem::Vector& elfun, mfem::DenseMatrix& elmat);
 
 private:
+  /**
+   * @brief Calculate the deformation gradient and right Cauchy-Green deformation tensor at a quadrature point
+   *
+   * @param[in] el The finite element
+   * @param[in] ip The integration point
+   * @param[in] Ttr The reference-to-target (stress-free) transformation
+   */
+  void CalcDeformationGradient(const mfem::FiniteElement& el, const mfem::IntegrationPoint& ip,
+                               mfem::ElementTransformation& Ttr);
+
   /**
    * @brief The associated hyperelastic model
    */
-  mfem::HyperelasticModel* model_;
+  serac::HyperelasticMaterial& material_;
+
+  /**
+   * @brief gradients of reference shape functions (dof x dim).
+   *
+   */
+  mfem::DenseMatrix DSh_;
+
+  /**
+   * @brief gradients of the shape functions in the target configuration (DN_i, DX_j)
+   *
+   */
+  mfem::DenseMatrix DS_;
```

Comment:
I'm used to "reference" being the undeformed mesh, but it looks like it might be referring to the master/parent element here?

I know mfem uses names like `DSh` and `DS` but those are pretty opaque. I'm trying out `dN_dx` for shape function gradient w.r.t. current configuration and `dN_dX` for shape function gradient w.r.t. reference configuration, and `dN_dxi` w.r.t. master/parent element in the variationalform branch.

Reply:
I'm a bit worried about having both `dN_dx` and `dN_dX` as variables since the only difference is case, but I agree that `DS` and `DSh` are not descriptive so I'll change those.

Reply:
So what we do is 

dN_dx0, dN_dxn, dN_dxnp1, dudXnphalf, etc.